systems/controllers/QPCommon.cpp

```diff
@@ -205,7 +239,26 @@ VectorXd velocityReference(NewQPControllerData *pdata, double t, const Ref<Vecto
     dt = t - pdata->state.t_prev;
   }
 
-  pdata->state.vref_integrator_state = (1-params->eta)*pdata->state.vref_integrator_state + params->eta*qd + qdd*dt;
+  VectorXd qdd_limited = qdd;
+  // Do not wind the vref integrator up against the joint limits for the legs
```

Comment:
Yeah, but the pots/encoders differences and need for high precision mean we do have to wind the arm integrators up against the joint stops, whereas for the legs it's not a good idea.